tests/copp/test_copp.py

```diff
@@ -98,6 +107,93 @@ def test_no_policer(self, protocol, duthosts, enum_rand_one_per_hwsku_frontend_h
+
+        logger.info("Verify {} trap status is installed by sending traffic".format(self.trap_id))
+        pytest_assert(
+            wait_until(60, 20, 0, _copp_runner, duthost, ptfhost, self.trap_id.upper(), copp_testbed, dut_type),
+            "Installing {} trap fail".format(self.trap_id))
+
+    @pytest.mark.parametrize("remove_trap_type", ["delete_feature_entry",
+                                                  "disable_feature_status"])
+    def test_remove_trap(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname, ptfhost, check_image_version, copp_testbed, dut_type, backup_restore_config_db, remove_trap_type):
+        """
+        Validates that The trap(bgp) can be uninstalled after deleting the corresponding entry from the feature table
+
+        1. Pre condition: make the tested trap installed and always_enable is false
+        2. Remove trap according to remove trap type
+        4. Verify the trap status is uninstalled by sending traffic
+        """
+        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+
+        logger.info("Pre condition: make trap {} is installed".format(self.feature_name))
+        pre_condition_install_trap(ptfhost, duthost, copp_testbed, self.trap_id, self.feature_name)
+
+        if remove_trap_type == "delete_feature_entry":
+            logger.info("Remove feature entry: {}".format(self.feature_name))
+            copp_utils.remove_feature_entry(duthost, self.feature_name)
+        else:
+            logger.info("Disable {} in feature table".format(self.feature_name))
+            copp_utils.disable_feature_entry(duthost, self.feature_name)
+
+        logger.info("Verify {} trap status is uninstalled by sending traffic".format(self.trap_id))
+        pytest_assert(
+            wait_until(100, 20, 0, _copp_runner, duthost, ptfhost, self.trap_id.upper(), copp_testbed, dut_type, has_trap=False),
+            "uninstalling {} trap fail".format(self.trap_id))
+
+    def test_trap_config_save_after_reboot(self, duthosts, localhost, enum_rand_one_per_hwsku_frontend_hostname, ptfhost,check_image_version, copp_testbed, dut_type, backup_restore_config_db):
+        """
+        Validates that the trap configuration is saved or not after reboot(reboot, fast-reboot, warm-reboot)
+
+        1. Set always_enabled of a trap(e.g. bgp) to true
+        2. Config save -y
+        3. Do reboot randomly(reboot/warm-reboot/fast-reboot/soft-reboot)
+        4. Verify configuration are saved successfully
+        5. Verify the trap status is installed by sending traffic
+        """
+
+        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+
+        logger.info("Set always_enabled of {} to true".format(self.trap_id))
+        copp_utils.configure_always_enabled_for_trap(duthost, self.trap_id, "true")
+
+        logger.info("Config save")
+        duthost.command("sudo config save -y")
+
+        reboot_type = random.choice(["cold", "fast", "warm", "soft"])
```

Comment:
Hi vcheketx,
Because reboot  consumes time very much, so we just want to choosing one randomly for one time. 
When we run the test for multiple times, all reboot-type will be covered. 
So we don't use test parametrization and parser.adoption.
thanks

Reply:
If it doesn't work for one reboot type, then the test case would be flaky and cause trouble.
Can you add an option to cover all reboot types? If the option is not supplied, just use one fixed default reboot type.

Reply:
Or add an option for specifying reboot type as vscheketx suggested?